firedrake/plot.py

```diff
@@ -1,373 +1,280 @@
-        plot(functions[val], num_points, ax, **kwargs)
-        plt.pause(0.01)
-    update(0)
-    func_slider.on_changed(update)
-
-    def auto_play(event):
-        if player.status == Player.PLAYING:
-            player.status = Player.STOP
-            return
-        curr = 0
-        player.status = Player.PLAYING
-        while curr < len(functions) and player.status == Player.PLAYING:
-            curr += 1
-            func_slider.set_val(curr)
-            plt.pause(player.frame_interval)
-        player.status = Player.STOP
-    play_button.on_clicked(auto_play)
-
-    def increase_speed(event):
-        player.frame_interval /= 2
-    plus_button.on_clicked(increase_speed)
-
-    def decrease_speed(event):
-        player.frame_interval *= 2
-    minus_button.on_clicked(decrease_speed)
-
-    return ax
-
-
-def plot_mesh(mesh, axes=None, surface=False, colors=None, **kwargs):
-    """Plot a mesh.
-
-    :arg mesh: The mesh to plot.
-    :arg axes: Optional matplotlib axes to draw on.
-    :arg surface: Plot surface of mesh only?
-    :arg colors: Colour for the edges (passed to constructor of LineCollection)
-    :arg **kwargs: Extra keyword arguments to pass to matplotlib.
-
-    Note that high-order coordinate fields are downsampled to
-    piecewise linear first.
-
+__all__ = ["plot", "triplot", "tricontourf", "tricontour", "trisurf", "tripcolor",
+           "quiver"]
+
+
+def _autoscale_view(axes, coords):
+    axes.autoscale_view()
+
+    # Dirty hack; autoscale_view doesn't appear to work for 3D plots.
+    if isinstance(axes, mpl_toolkits.mplot3d.Axes3D):
+        setters = ["set_xlim", "set_ylim", "set_zlim"]
+        for setter, idx in zip(setters, range(coords.shape[1])):
+            try:
+                setter = getattr(axes, setter)
+            except AttributeError:
+                continue
+            amin = coords[:, idx].min()
+            amax = coords[:, idx].max()
+            extra = (amax - amin) / 20
+            if extra == 0.0:
+                # 1D interval
+                extra = 0.5
+            amin -= extra
+            amax += extra
+            setter(amin, amax)
+
+
+def _get_collection_types(gdim, tdim):
+    if gdim == 2:
+        if tdim == 1:
+            # Probably a CircleCollection?
+            raise NotImplementedError("Didn't get to this yet...")
+        elif tdim == 2:
+            return LineCollection, PolyCollection
+    elif gdim == 3:
+        if tdim == 1:
+            raise NotImplementedError("Didn't get to this one yet...")
+        elif tdim == 2:
+            return Line3DCollection, Poly3DCollection
+        elif tdim == 3:
+            return Poly3DCollection, Poly3DCollection
+
+    raise ValueError("Geometric dimension must be either 2 or 3!")
+
+
+def triplot(mesh, axes=None, interior_kw={}, boundary_kw={}):
+    r"""Plot a mesh with a different color for each boundary segment
+
+    :arg mesh: mesh to be plotted
+    :arg axes: axis object on which to plot mesh
```

Comment:
The `Axes` object note would ideally link to the matplotlib API documentation

Reply:
You're right about the naming (axes/ax) -- the current Firedrake plotting API uses "axes", so I stuck to that convention rather than change it to the one used in matplotlib. I can see your case for changing it to "ax" though. @wence- , any thoughts?

Reply:
Thinking about it, we've already deviated from matplotlib's `triplot` API by specifying the `Axes` as a keyword argument instead of the default first argument. I therefore lean towards maintaining the firedrake naming to avoid confusion and to maintain consistent naming within our own API. The documentation for `Firedrake.triplot` makes it clear what this refers to.

@danshapero if you agree then feel free to resolve this.